code/centauri/algorithms/grandpa/verifier/src/lib.rs

```diff
@@ -69,18 +69,19 @@ where
 		Err(anyhow!("Latest finalized block should be highest block in unknown_headers"))?;
 	}
 
-	let from = finality_proof
-		.unknown_headers
-		.iter()
-		.min_by_key(|h| *h.number())
-		.ok_or_else(|| anyhow!("Unknown headers can't be empty!"))?;
-	let mut finalized = headers
-		.ancestry(from.hash(), target.hash())
-		.map_err(|_| anyhow!("Invalid ancestry!"))?;
+	let from = client_state.latest_relay_hash;
+	let mut finalized = headers.ancestry(from, target.hash()).map_err(|_| {
+		anyhow!("[verify_parachain_headers_with_grandpa_finality_proof] Invalid ancestry!")
+	})?;
 	finalized.sort();
 
 	// 2. verify justification.
 	let justification = GrandpaJustification::<H>::decode(&mut &finality_proof.justification[..])?;
+
+	if justification.commit.target_hash != finality_proof.block {
```

Comment:
rather than having to do this check everywhere, let's just remove `block` from `FinalityProof` and use the `justification.commit.target_hash` everywhere instead

Reply:
In some places (like [here](https://github.com/ComposableFi/composable/blob/centauri/code/centauri/light-clients/ics10-grandpa/src/client_def.rs#L148)) we don't decode justification and use this field. How about using the last item in `unknown_headers`?

Reply:
Ok as long as we enforce in the verifier that 

```rust
unkown_headers.iter().max_by_key(Header::number) == finality_proof.block
```

Reply:
Yes, I think we should enforce that since it's one of the [invariants](https://github.com/ComposableFi/composable/blob/centauri/code/centauri/algorithms/grandpa/primitives/src/lib.rs#L55)